src/nvim/eval.c

```diff
@@ -10764,32 +10764,40 @@ static void f_getcompletion(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     options |= WILD_KEEP_ALL;
   }
 
+  if (argvars[0].v_type != VAR_STRING || argvars[1].v_type != VAR_STRING) {
+    EMSG(_(e_invarg));
+    return;
+  }
+
   ExpandInit(&xpc);
   xpc.xp_pattern = get_tv_string(&argvars[0]);
-  xpc.xp_pattern_len = (int)STRLEN(xpc.xp_pattern);
-  xpc.xp_context = cmdcomplete_str_to_type(get_tv_string(&argvars[1]));
-  if (xpc.xp_context == EXPAND_NOTHING) {
-    if (argvars[1].v_type == VAR_STRING) {
+  xpc.xp_pattern_len = STRLEN(xpc.xp_pattern);
```

Comment:
this is immediately overwritten by the if{} block. Can it be moved to the else{} block?